clang/docs/SanitizerSpecialCaseList.rst

```diff
@@ -48,6 +49,59 @@ Example
   $ clang -fsanitize=address -fsanitize-ignorelist=ignorelist.txt foo.c ; ./a.out
   # No error report here.
 
+Usage with UndefinedBehaviorSanitizer
+=====================================
+
+The arithmetic overflow sanitizers ``unsigned-integer-overflow`` and
+``signed-integer-overflow`` as well as the implicit integer truncation
+sanitizers ``implicit-signed-integer-truncation`` and
+``implicit-unsigned-integer-truncation`` support the ability to adjust
+instrumentation based on type.
+
+.. code-block:: bash
+
+  $ cat foo.c
+  void foo() {
+    int a = 2147483647; // INT_MAX
+    ++a;                // Normally, an overflow with -fsanitize=signed-integer-overflow
+  }
+  $ cat ignorelist.txt
+  [signed-integer-overflow]
+  type:int
+  $ clang -fsanitize=signed-integer-overflow -fsanitize-ignorelist=ignorelist.txt foo.c ; ./a.out
+  # no signed-integer-overflow error
+
+For example, supplying the above ``ignorelist.txt`` to
+``-fsanitize-ignorelist=ignorelist.txt`` disables overflow sanitizer
+instrumentation for arithmetic operations containing values of type ``int``.
+
+The following SCL categories are supported: ``=no_sanitize`` and ``=sanitize``.
+The ``no_sanitize`` category is the default for any entry within an ignorelist
```

Comment:
It need to clarify that intuitive order does not work. E.g. user may expect that the last marcher is what is applied to the type.

If fact it's more complicated, I even can't guess the rule without reading the implementation.

Reply:
Actually 'SpecialCaseList::inSectionBlame' supports linenumber
So we can make it work, with some refactoring, in future patches.

Maybe put comment here that order if undefined?

SpecialCaseList supports multiple files as input, comparing linenumber between them is not helpfull, should be a pair <> of numbers.

Reply:
> Actually 'SpecialCaseList::inSectionBlame' supports linenumber
> So we can make it work, with some refactoring, in future patches.

Looking trough code, I think the rule is simple any match of "sanitize" > "no_sanitize". Order is irrelevant.
I guess we can keep this way.
Please document.

Please add a test with different permutations, as I commented below.

Reply:
Done in https://github.com/llvm/llvm-project/pull/107332/commits/33aa461aeaea6e42aea80745b5bf3fbeab6c5499

Thanks